packages/ruleset/src/utils/validate-subschemas.js

```diff
@@ -0,0 +1,36 @@
+const validateSubschemas = (schema, path, validate) => {
```

Comment:
How confident are we in the term "subschemas" here? Should we reconsider it or at least define it in a header comment? Perhaps:

```
Subschemas include property schemas (for an object schema), item schemas (for an array schema), and applicator schemas (such as those in an `allOf`, `anyOf`, or `oneOf` property), plus all subschemas _of those schemas_.
```

Reply:
It was the first thing that came to me so I'm not particularly attached to it but I feel it describes the intent (all schemas "below" the current schema). I'm open to suggestions, though. And I like the idea of adding that comment, I can make that change

Reply:
I think the definition I proposed above is consistent with how JSON Schema itself uses the word but I still think it might be worth defining our own use.

Reply:
"validateSchemaSubTree()" :)